consensus/src/processes/window.rs

```diff
@@ -686,4 +711,24 @@ impl BoundedSizeBlockHeap {
         self.binary_heap.push(r_sortable_block);
         true
     }
+
+    // This method is intended to be used to merge the ancestor heap with the current heap.
+    // It is expected that any ancestor heap will have lower blue work blocks than the current heap.
+    fn merge_ancestor_heap(&mut self, ancestor_heap: &mut BlockWindowHeap) {
+        // because popping has O(log(n)) complexity, we pop the excess amount first, then append the two.
+        let overflow_amount = (self.len() + ancestor_heap.len()) - self.size_bound;
+        if overflow_amount > 0 {
```

Comment:
You don't need this `if` condition. `overflow_amount` will always be either 0 or exactly the overflow amount. (Also if it were negative, the above would panic without the saturating sub change).

Reply:
I do find the explicit > 0 check to be more readable, but yes iterating form 0 to 0 should be a no-op i guess, might just add a comment in then.